### ISSUE TITLE
[Good First Issue][TF FE]: Support RGBToHSV operation for TensorFlow

### ISSUE THREAD

**Issue by rkazants** [openvinotoolkit/openvino#24792]
### Context

OpenVINO component responsible for support of TensorFlow models is called as TensorFlow Frontend (TF FE). TF FE converts a model represented in [TensorFlow opset](https://www.tensorflow.org/api_docs/python/tf/raw_ops) to a model in [OpenVINO opset](https://docs.openvino.ai/2023.2/openvino_docs_ops_opset13.html).

In order to infer TensorFlow models with [RGBToHSV](https://www.tensorflow.org/api_docs/python/tf/raw_ops/RGBToHSV) operation by OpenVINO, TF FE needs to be extended with this operation support.

### What needs to be done?

For [RGBToHSV](https://www.tensorflow.org/api_docs/python/tf/raw_ops/RGBToHSV) operation support, you need to implement the corresponding loader into TF FE [op](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/tensorflow_common/src/op) directory and to register it into the dictionary of [Loaders](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/src/op_table.cpp). One loader is responsible for conversion (or decomposition) of one type of TensorFlow operation.

Here is an example of loader implementation for TensorFlow `Einsum` operation:

```cpp
OutputVector translate_einsum_op(const NodeContext& node) { 
     auto op_type = node.get_op_type(); 
     TENSORFLOW_OP_VALIDATION(node, op_type == "Einsum", "Internal error: incorrect usage of translate_einsum_op."); 
     auto equation = node.get_attribute<std::string>("equation"); 
  
     OutputVector inputs; 
     for (size_t input_ind = 0; input_ind < node.get_input_size(); ++input_ind) { 
         inputs.push_back(node.get_input(input_ind)); 
     } 
  
     auto einsum = make_shared<Einsum>(inputs, equation); 
     set_node_name(node.get_name(), einsum); 
     return {einsum}; 
 } 
```

In this example, `translate_einsum_op` converts TF `Einsum` into OV `Einsum`. `NodeContext` object passed into the loader packs all info about inputs and attributes of `Einsum` operation. The loader retrieves an attribute of the equation by using the `NodeContext::get_attribute()` method, prepares input vector, creates `Einsum` operation from OV opset and returns a vector of outputs.

Responsibility of a loader is to parse operation attributes, prepare inputs and express TF operation via OV operations sub-graph. Example for `Einsum` demonstrates the resulted sub-graph with one operation. In PR https://github.com/openvinotoolkit/openvino/pull/19007 you can see operation decomposition into multiple node sub-graph.

Once you are done with implementation of the translator, you need to implement the corresponding layer tests `test_tf_AdjustHue.py` and put it into [layer_tests/tensorflow_tests](https://github.com/openvinotoolkit/openvino/tree/master/tests/layer_tests/tensorflow_tests) directory. Example how to run some layer test:

```
export TEST_DEVICE=CPU
cd openvino/tests/layer_tests/tensorflow_tests
pytest test_tf_Shape.py
```

### Example Pull Requests

https://github.com/openvinotoolkit/openvino/pull/24511

### Hint

Check this out: https://github.com/openvinotoolkit/openvino/pull/24511

### Resources

- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [How to Build OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build.md)
- [Developer documentation for TensorFlow Frontend](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/tensorflow)
- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/7pVRxUwdWG) - engage in discussions, ask questions and talk to OpenVINO developers

### Contact points

- @openvinotoolkit/openvino-tf-frontend-maintainers
- @rkazants in GitHub
- rkazants in Discord

**Comment by sumwun1** [comment]
I've never contributed to an open source project before (besides fixing typos in documentation), but I do have 2 or 3 years of C++ experience and I've read this repository's contributing.md. Can I assign this to myself and start working?

**Comment by duydl** [comment]
@sumwun1 go ahead. The main converting logic is already implemented. You would only need to check how to add new TF FE ops (see Change Files in relevant PR)

**Comment by rkazants** [comment]
Hi @sumwun1, the task is your yours. Please proceed with execution:) 
Pay attention to that we anticipate PR for 2 weeks after which we release GFI for other people if no progress.

Best regards,
Roman

**Comment by sumwun1** [comment]
This has nothing to do with this particular issue, but is there some way to get notifications for every comment that either mentions my username or is a response to my comments? After I posted on this thread, I was able to find it again only because I searched my username in the search bar.

**Comment by sumwun1** [comment]
@rkazants Where can I find documentation for classes like NodeContext or functions like make_shared? Those are the things I was supposed to use for this issue, right?

**Comment by rkazants** [comment]
> @rkazants Where can I find documentation for classes like NodeContext or functions like make_shared? Those are the things I was supposed to use for this issue, right?

`NodeContext` class belongs to developer API so you can find documentation in the code: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/common/include/openvino/frontend/node_context.hpp

`make_shared` is a standard C++ API to work with smart pointers.

Best regards,
Roman

**Comment by sumwun1** [comment]
How do I know which functions belong to which APIs?

**Comment by sumwun1** [comment]
Where's the documentation for ov::frontend::tensorflow::create_same_type_const_scalar? I tried searching Google and Openvino's documentation site and couldn't find it.

**Comment by sumwun1** [comment]
> Where's the documentation for ov::frontend::tensorflow::create_same_type_const_scalar? I tried searching Google and Openvino's documentation site and couldn't find it.

@rkazants ?

**Comment by sumwun1** [comment]
@rkazants None of the tests are passing on my computer, even the ones from pull requests that were already merged, which makes me think I accidentally set up pytest wrong. How do I read the test output to see what's wrong? Here's one of the outputs.
[HSVToRGB.txt](https://github.com/user-attachments/files/15845427/HSVToRGB.txt)

**Comment by rkazants** [comment]
> @rkazants None of the tests are passing on my computer, even the ones from pull requests that were already merged, which makes me think I accidentally set up pytest wrong. How do I read the test output to see what's wrong? Here's one of the outputs. [HSVToRGB.txt](https://github.com/user-attachments/files/15845427/HSVToRGB.txt)

Hi @sumwun1, responded in Discord.
Please be informed that we have to release GFI in ~2 weeks if we see no progress and assign to other person. This GFI has a good reference for moving forward.

Best regards,
Roman

**Comment by sumwun1** [comment]
> Hi @sumwun1, responded in Discord. Please be informed that we have to release GFI in ~2 weeks if we see no progress and assign to other person. This GFI has a good reference for moving forward.
> 
> Best regards, Roman

This is the fork I've been using for this issue. You can see the progress I've made on there.
https://github.com/sumwun1/openvino

**Comment by sumwun1** [comment]
https://drive.google.com/file/d/1te31UzQAmZQCsEkMp1Yi8yKqHTWubZ45/view?usp=sharing
I got this absolutely massive error message when running "cmake --build . --config Release --verbose". I used the Windows 10 terminal instead of WSL this time. What can I do to try to fix it? Would it help if I include the number of jobs? If yes, how many jobs should I specify?

**Comment by rkazants** [comment]
Hi @sumwun1,

I generate `.sln` file and build OV inside MSVS:

```
cmake -G "Visual Studio 16 2019" -A x64 -DCMAKE_BUILD_TYPE=Release -DENABLE_TESTS=ON ..
```

Best regards,
Roman

**Comment by sumwun1** [comment]
I got an error saying "CMake Error: The source directory "D:/src" does not appear to contain CMakeLists.txt." (D:/src is the folder that contains my copy of Openvino) Should I put CMakeLists.txt in that folder or somehow make Cmake look for the file in a different folder? 

I also tried following the Linux build instructions using WSL, and I was unable to run "sudo ./install_build_dependencies.sh" even though the file was in my working directory. How can I fix this?
